src/maidsafe/routing/connection_manager.cc

```diff
@@ -204,39 +207,42 @@ void ConnectionManager::InsertPeer(PeerNode&& node_arg) {
 void ConnectionManager::StartReceiving(PeerNode& node) {
   auto node_guard = node.DestroyGuard();
 
-  node.Receive([=, &node](asio::error_code error, const Bytes& bytes) {
-      if (!node_guard.lock()) return;
-      if (error) return;
-      if (!on_receive_) return;
-      // Complex handler invocation to be safe in cases where the
-      // handler destroys this object or in case where the handler
-      // invocation resets the handler to something else.
-      auto h = move(on_receive_);
-      h(node.id(), bytes);
-      if (!node_guard.lock()) return;
-      if (!on_receive_) {
-        on_receive_ = move(h);
-      }
-      StartReceiving(node);
-      });
-
+  node.Receive([=, &node](asio::error_code error, const SerialisedMessage& bytes) {
+    if (!node_guard.lock())
+      return;
+    if (error)
+      return;
+    if (!on_receive_)
+      return;
+    // Complex handler invocation to be safe in cases where the
+    // handler destroys this object or in case where the handler
+    // invocation resets the handler to something else.
+    auto h = move(on_receive_);
+    h(node.id(), bytes);
+    if (!node_guard.lock())
+      return;
+    if (!on_receive_) {
+      on_receive_ = move(h);
+    }
+    StartReceiving(node);
+  });
 }
 
-//bool ConnectionManager::CloseGroupMember(const Address& their_id) {
+// bool ConnectionManager::CloseGroupMember(const Address& their_id) {
 //  auto close_group(routing_table_.OurCloseGroup());
 //  return std::any_of(std::begin(close_group), std::end(close_group),
 //                     [&their_id](const NodeInfo& node) { return node.id == their_id; });
 //}
 
 optional<CloseGroupDifference> ConnectionManager::GroupChanged() {
-  auto new_nodeinfo_group(OurCloseGroup());
-  std::vector<Address> new_group;
-  for (const auto& nodes : new_nodeinfo_group)
-    new_group.push_back(nodes.id);
-
-  if (new_group != current_close_group_) {
-    auto changed = std::make_pair(new_group, current_close_group_);
-    current_close_group_ = new_group;
+  auto new_group(OurCloseGroup());
+  std::vector<Address> new_group_ids;
+  for (const auto& group_member_public_pmid : new_group)
+    new_group_ids.push_back(Address(group_member_public_pmid.name()->string()));
+
+  if (new_group_ids != current_close_group_) {
+    auto changed = std::make_pair(new_group_ids, current_close_group_);
+    current_close_group_ = new_group_ids;
     return changed;
```

Comment:
I think this function together with the CloseGroupDifference can go away completely as the new Connections class will notify it's users about connectivity changes one by one.